packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md

```diff
@@ -209,31 +231,19 @@ events.
 
-## X API
+#### X API
 
-The X API allows your website to communicate with Interface X. It is a set of utilities that helps
-to integrate Interface X into your website.
+The [X&nbsp;API](https://github.com/empathyco/x/blob/main/packages/x-components/src/x-installer/api/base-api.ts) object allows your commerce store to communicate with Interface&nbsp;X. It supports multiple functions to integrate Interface&nbsp;X in your website.
 
 | Function         | Parameters                                                                     | Description                                                                                                                                                       |
 | ---------------- | ------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| init             | - [Snippet Configuration](#snippet-configuration) - The initialisation options | [Initialises Interface X on demand](#initialise-on-demand).                                                                                                       |
-| search           | - query (Optional) - The query to open Interface&nbsp;X with                   | Opens Interface&nbsp;X and triggers a search with the given query.                                                                                                |
-| setSnippetConfig | - [Snippet Configuration](#snippet-configuration) - The Initialisation options | Changes the options and all components react to the changes. Useful, for example, to change both search engine and displayed language without reloading the page. |
+| `init`             | [snippet configuration params](#snippet-configuration) - Initialization options | [Initializes Interface&nbsp;X on demand](#initializing-interface-x-project-on-demand).   |
+| `search`           | `query` - Query to open Interface&nbsp;X  | _Optional_. Executes Interface&nbsp;X and triggers a search with the given query.  |
+| `setSnippetConfig` | [snippet configuration params](#snippet-configuration) - Initialization options | Changes initialization options so that all components react to the changes, i.e. changing both search engine and language without reloading the page. |
+| `addProductToCart` | none | Sends tracking of the `AddToCartevent` to the [Empathy Tagging microservice](https://docs.empathy.co/develop-empathy-platform/capture-interaction-signals/tagging-api-guide.html) for the product displayed on screen. This function is called from  the product detail page (PDP) when the shopper clicks on the add-to-cart button. |
```

Comment:
```suggestion
| `addProductToCart` | `productId` __Optional__ - The product identifier that has been added to the cart  | Sends tracking of the `AddToCart` event to the [Empathy Tagging microservice](https://docs.empathy.co/develop-empathy-platform/capture-interaction-signals/tagging-api-guide.html) for the product displayed on screen. This function is called from  the product detail page (PDP) when the shopper clicks on the add-to-cart button. If the `productId` is not provided, the URL will be used to detect if the user has discovered the product thanks to a search session. |
```

Reply:
Just wondering @herrardo @javieri-empathy, if the parameters for `AddProductToCart` and `Search` functions are optional, then do I guess that parameters for other functions (`setSnippetConfig` or `Init`) are required?? If so, should we say which are required rather than the optional ones? I'm just trying to unify styles, in the Config snippet table and the Props/Slots in UI ref, we indicate the required ones only.

Reply:
My answer is going to hurt you. I would inform every one of them with _Optional_ or _Required_because when it was a check on a column, it was obvious, but now it is not so.

Anyway, if you inform only about the Required ones is not a stopper for me.